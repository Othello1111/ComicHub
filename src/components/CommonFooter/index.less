@import '~antd/lib/style/themes/default.less';

.container {
    text-align: center;
}
