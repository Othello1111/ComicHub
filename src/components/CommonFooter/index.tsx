import React from 'react';
import styles from './index.less';

const CommonFooter: React.FunctionComponent = () => (
  <footer className={styles.container}>
    <span>Copyright&nbsp;@</span>
    &nbsp;
    <a href="https://github.com/nusr" target="_blank" rel="noopener noreferrer">
      nusr
    </a>
  </footer>
);
export default CommonFooter;
