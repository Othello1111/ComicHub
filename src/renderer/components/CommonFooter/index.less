@import '~antd/lib/style/themes/default.less';

.container {
  padding: 24px;
  text-align: center;
  background-color: #fff;
  border-top: 1px solid @border-color-base;
}
