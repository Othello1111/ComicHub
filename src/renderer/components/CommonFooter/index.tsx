import React from 'react';
import styles from './index.less';

const CommonFooter: React.FunctionComponent = () => {
    return (
        <footer className={styles.container}>
            Copyright @
            <a href="https://github.com/nusr" target="_blank">
                nusr
            </a>
        </footer>
    );
};
export default CommonFooter;
