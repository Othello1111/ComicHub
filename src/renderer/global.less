@import '~antd/lib/style/themes/default.less';

body {
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #000;
  line-height: 1.5;
  background-color: #f2f4f5;
}
