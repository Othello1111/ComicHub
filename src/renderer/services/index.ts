import request from '../utils/request';

interface PostData {
    url?: string;

    [index: string]: string;
}

export async function getMenuList() {
    return request('/v1/menu');
}

export async function postItem(params: PostData = {}) {
    const { url, ...rest } = params;
    if (!url) {
        return;
    }
    const realUrl = `/v1/${url}`;
    return request(realUrl, {
        method: 'POST',
        body: rest,
    });
}
