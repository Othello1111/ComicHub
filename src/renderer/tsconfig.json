{
    "compilerOptions": {
        "target": "esnext",
        "module": "esnext",
        "moduleResolution": "node",
        "importHelpers": true,
        "jsx": "preserve",
        "esModuleInterop": true,
        "sourceMap": true,
        "strict": true,
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "baseUrl": ".",
        "paths": {
            "@/*": ["./*"]
        },
        "allowSyntheticDefaultImports": true
    }
}
