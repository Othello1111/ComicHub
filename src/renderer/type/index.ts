export interface SharedState {
    currentUrl: string;
    currentType: string;
}
export interface MenuItem {
    value: string;
    enabled: boolean;
    base: string;
    name: string;
}
export interface IFormData {
    url: string;
    name: string;
}

export interface IOptionData {
    value: string | number;
    name: string;
    enabled: boolean;
}
