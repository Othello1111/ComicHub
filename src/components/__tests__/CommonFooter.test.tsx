import React from 'react';
import renderer from 'react-test-renderer';
import CommonFooter from '../CommonFooter';

it('Component: CommonFooter Snapshots', () => {
  const tree = renderer.create(<CommonFooter/>).toJSON();
  expect(tree).toMatchSnapshot();
});

