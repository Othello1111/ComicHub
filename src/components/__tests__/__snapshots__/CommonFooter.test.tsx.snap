// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CommonFooter Snapshots 1`] = `
<footer
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <span>
    Copyright @
  </span>
   
  <a
    href="https://github.com/nusr"
    rel="noopener noreferrer"
    target="_blank"
  >
    nusr
  </a>
</footer>
`;
